### ISSUE TITLE
PyTorch 2.0 F.scaled_dot_product_attention issue

### ISSUE THREAD

**Issue by glistering96** [glistering96/AlphaRouter#7]
The recent function torch.nn.functional.scaled_dot_product_attention is ONLY VALID WHEN THE MASKING IS NOT SET.

When we throw a mask explicitly with the function, e.g., F.scaled_dot_product_attention(Q, K, V, attn_mask=mask) with some float mask tensor, it throws an error message `"Both fused kernels do not support non-null attn_mask."`

which is s defined in the original src code of PyTorch in aten/src/ATen/native/transformers/cuda/sdp_utils.cpp on line 267 as like

```
bool check_for_attn_mask(sdp_params params, bool debug) {
  if (params.has_attn_mask) {
    if (debug) {
      TORCH_WARN("Both fused kernels do not support non-null attn_mask.");
    }
    return false;
  }
  return true;
}
```

the `params` input which is of the `sdp_params` type is defined in aten/src/ATen/native/transformers/cuda/sdp_utils.h as a boolean type of struct.

This struct is used in the code aten/src/ATen/native/transformers/cuda/attention.cu on line 571 as 

` sdp::sdp_params kernel_params{q, k, v, mask.has_value(), 0.0, false};`

Here, `mask` is thought to be an instance of some pytorch tensor class for C++, and found that this `mask.has_value()` basically returns true if the tensor exists on memory and false if not.

So the warning "Both fused kernels do not support non-null attn_mask." seems to be some warning related to saying that users should not use a masking tensor with `scaled_dot_product_attention`.
